knime_extension/src/nodes/spatialmodels.py

```diff
@@ -1716,3 +1716,1279 @@ def execute(self, exec_context: knext.ExecutionContext, input_1, input_2):
             knext.Table.from_pandas(results),
             knext.view_html(html),
         )
+
+
+# New added from Geolab in 2023-11-20
+
+SHOULD_NOT_CONTAIN_MISSING_VALUES = """
+   **Note:** The input table should not contain missing values. You can use the
+    [Missing Value](https://hub.knime.com/knime/extensions/org.knime.features.base/latest/org.knime.base.node.preproc.pmml.missingval.compute.MissingValueHandlerNodeFactory/) node to replace them."""
+
+# Root path for all node icons in this file
+__NODE_ICON_PATH_2 = "icons/icon/Geolab/"
+
+############################################
+# spatial GM_Error node
+############################################
+
+
+@knext.node(
+    name="Spatial GM Error",
+    node_type=knext.NodeType.LEARNER,
+    category=__category,
+    icon_path=__NODE_ICON_PATH_2 + "GMErr.png",
+)
+@knext.input_table(
+    name="Input Table",
+    description="Input Table with dependent and independent variables for calculation of the spatial GM_Error model.",
+)
+@knext.input_table(
+    name="Spatial Weights",
+    description="Input Table with spatial weights for calculation of the spatial GM Error model.",
+)
+@knext.output_table(
+    name="Output Table",
+    description="Description of the spatial GM Error model.",
+)
+@knext.output_table(
+    name="Variable and Coefficient Table",
+    description="Variable and Coefficient Table of the spatial GM Error model.",
+)
+@knext.output_view(
+    name="Model summary view",
+    description="Model summary view of the spatial GM Error model.",
+)
+class SpatialGM_Error:
+    (
+        """
+    Spatial GM Error Model.
+    GMM method for a spatial error model, with results and diagnostics; based on Kelejian and Prucha (1998, 1999) [KP98] [KP99]. More information can be found at [here](https://spreg.readthedocs.io/en/latest/generated/spreg.GM_Error.html#spreg.GM_Error). Please refer the following papers for more details.
+    - %s
+    - %s
+    """
+        % (mrs.model_references["KP98"], mrs.model_references["KP99"])
+        + SHOULD_NOT_CONTAIN_MISSING_VALUES
+    )
+
+    geo_col = knut.geo_col_parameter()
+
+    id_col = get_id_col_parameter()
+
+    dependent_variable = knext.ColumnParameter(
```

Comment:
Dependent variable and independent variables are used in most of the nodes. Wouldn't it make sense to extract their definition into functions?